plugins/TagFix_DuplicateValue.py

```diff
@@ -99,37 +115,45 @@ def levenshtein(self, s1, s2):
+        for r in set_of_regexes:
+            if r.match(key):
+                return True
+        return False
+
     def node(self, data, tags):
         err = []
         keys = tags.keys()
-        for k in keys:
-            if k in self.BlackList:
-                continue
-
-            try:
-                for blr in self.BlackListRegex:
-                    if blr.match(k):
-                        raise Exception
-            except Exception:
-                continue
 
+        for k in keys:
+            if k in self.WhitelistSimilarEqual:
+                continue # Key may have equal or similar values
             v = tags[k]
+            if not ';' in v:
+                continue
+            if self.anyRegexMatch(self.WhitelistSimilarEqualRegex, k):
+                continue # Key may have equal or similar values
             if k == 'source':
                 v = v.replace('Cadastre ; mise', 'Cadastre, mise') # France
                 v = v.replace('GSImaps/ort', 'GSImaps/std') # Japan
-            if ';' in v:
-                vs = list(filter(lambda w: len(w) > 0, map(lambda w: w.strip(), v.split(';'))))
-                if len(vs) != len(set(vs)):
-                    err.append({"class": 3060, "subclass": stablehash64(k),
-                                "text": T_("Duplicated values {key}={val}", key = k, val = tags[k]),
-                                "fix": {k: ";".join(set(vs))} })
-                else:
-                    vs_long = filter(lambda w: len(w) > 6, vs)
-                    for v1,v2 in itertools.combinations(vs_long, 2):
-                        if abs(len(v1)-len(v2)) < 4 and self.levenshtein(v1, v2) < 4:
-                            err.append({"class": 30601, "subclass": stablehash64(k),
-                                        "text": T_("Duplicated similar values {key}={val}", key = k, val = tags[k])})
-                            break
+            vs = list(filter(lambda w: len(w) > 0, map(lambda w: w.strip(), v.split(';'))))
+
+            if len(vs) != len(set(vs)):
+                err.append({"class": 3060, "subclass": stablehash64(k),
+                            "text": T_("Duplicated values {key}={val}", key = k, val = tags[k]),
+                            "fix": {k: ";".join(set(vs))} })
+            else:
+                if k in self.WhitelistSimilar:
+                    continue # Key may have similar values
+                if self.anyRegexMatch(self.WhitelistSimilarRegex, k):
+                    continue # Key may have similar values
+
+                vs_long = filter(lambda w: len(w) > 6, vs)
+                for v1,v2 in itertools.combinations(vs_long, 2):
+                    if abs(len(v1)-len(v2)) < 4 and self.levenshtein(v1, v2) < 4:
+                        err.append({"class": 30601, "subclass": stablehash64(k),
+                                    "text": T_("Duplicated similar values {key}={val}", key = k, val = tags[k])})
```

Comment:
Yes,  you not changed this, but could be better.
```suggestion
                                    "text": T_("Similar values {v1} and {v2} in {key}={val}", v1 = v1, v2 = v2, key = k, val = tags[k])})
```

Reply:
I like it :)
It may be that the message sometimes becomes too long; in that case we should probably drop the `={val}` at the end, but lets see